pkg/proxy/apis/kubeproxyconfig/validation/validation.go

```diff
@@ -238,3 +240,24 @@ func validateIPVSSchedulerMethod(scheduler kubeproxyconfig.IPVSSchedulerMethod,
 	}
 	return allErrs
 }
+
+func validateKubeProxyNodePortAddress(nodePortAddresses []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	for i := range nodePortAddresses {
+		if nodePortAddresses[i] == string(kubeproxyconfig.DefaultRoute) {
+			// special valid cases
+			continue
+		}
+		if nodePortAddresses[i] == "" {
+			allErrs = append(allErrs, field.Invalid(fldPath, nodePortAddresses, "must be non-empty"))
+			break
+		}
+		if _, _, err := net.ParseCIDR(nodePortAddresses[i]); err != nil {
```

Comment:
We should probably allow bare IPs, too?

Reply:
bare IPs can be expressed by CIDR. For example,

1.2.3.4 can be expressed by 1.2.3.4/32. In this way, we can keep `GetNodeAddresses()` cleaner and simpler. What do you think about it?

Reply:
OK for now.  The UX is a little weird, but probably won't get that much use.